sycl/test-e2e/bindless_images/sampled_fetch/fetch_2D.cpp

```diff
@@ -0,0 +1,125 @@
+// REQUIRES: linux
+// REQUIRES: cuda
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+// Uncomment to print additional test information
+// #define VERBOSE_PRINT
+
+class kernel_sampled_fetch;
+
+int main() {
+
+  sycl::device dev;
+  sycl::queue q(dev);
+  auto ctxt = q.get_context();
+
+  // Check if device supports 2D non-USM sampled image fetches
+  if (!dev.has(sycl::aspect::ext_oneapi_bindless_sampled_image_fetch_2d)) {
+#ifdef VERBOSE_PRINT
```

Comment:
Useful information should not be optional. When running this through LIT, the output will only be printed if the test failed, in which case we would probably want the verbose information anyway.

Reply:
With the aspects being checked by LIT through `// REQUIRES: aspect-<...>` this chunk of code is obsolete and I've removed it from the tests.

Reply:
My comment applies to the other cases where `VERBOSE_PRINT` is used. In the case that this test fails, it would be much more useful if we can just go look at the logs rather than having to change and recompile the test locally to get it.

Reply:
If `VERBOSE_PRINT` was always on, and some test fails, that test alone could print anywhere from tens of thousands to millions of lines with mismatches. Most of the time the information that it prints is not all that useful by itself, but rather used in the process of debugging, e.g. if the output we get back is all zeros, we can change the code in libclc to return a constant value to check if the correct intrinsic is even being called, or to trace where exactly an incorrect call is being made in the chain of calls from the API down to the intrinsics.

I don't think spewing out thousands to millions of lines per failed test would be that useful, and a lot of the time it would print beyond the limit of how many lines a terminal can show, In that case as part of the debugging process we would reduce the size of the image then turn on `VERBOSE_PRINT` so that the number of printed lines is a reasonable couple hundred instead of millions. So for those reasons I don't think it is useful to have this on all the time, it is a debugging tool that has to be used alongside other changes to the tests to become useful. Does that sound reasonable?

Reply:
At that point, I don't even know why we keep it in. If we are printing millions of lines of debugging information, it sounds like the use-case of the information is too broad to be useful for the regular developer. If so, people debugging it may as well just write their own temporary debug printing that suits their need.

Reply:
Well part of the debugging process is reducing the size of the image so that instead of 1024 x 1024 lines of mismatches, we print 16 x 16, and then uncommenting the `VERBOSE_PRINT` means we already have the debug printing code ready. This is much faster than writing temporary debug printing code each time something goes wrong. We expect we'll to use the verbose print when we add new backends (HIP, L0). Perhaps once the extension is less experimental and more stable with more backends, we can then remove all the debug printing code. At the moment it is still a useful tool in our development of the extension, and much more convenient than writing temporary debug printing code for every test that fails.

Reply:
I wonder if we can connect `VERBOSE_PRINT` to some lit argument or something. We could then automatically pass a define as part of `%{build}` shortcut and no source code changes would be needed to run "debugging" version of the test, just `llvm-lit --regular --args /path/to/test --verbose-print-or-something`.

Reply:
I've implemented a lit argument as you suggested @AlexeySachkov. Now the user only has to pass a lit parameter `verbose-print` through lit's `--param` or the equivalent `-D` options.

Specifying `--param verbose-print` will now add a `VERBOSE_PRINT` macro to the `%{build}` command.

Is this solution satisfactory? @steffenlarsen

Reply:
Yep! I'm okay with that, thanks! 😄